src/PKSim.Presentation/Presenters/Simulations/PopulationPKAnalysisPresenter.cs

```diff
@@ -68,23 +69,26 @@ public void CalculatePKAnalyses(IPopulationDataCollector populationDataCollector
          //Calculate first global PK that should always be updated
          _globalPKAnalysisPresenter.CalculatePKAnalysis(new[] { simulation });
 
-         var pkParameters = extractPKParameters(populationAnalysis, simulation);
-         if (!pkParameters.Any())
+         var pkParametersCache = extractPKParameters(populationAnalysis, simulation);
+         if (!pkParametersCache.Any())
             return;
 
-         var captionPrefix = populationAnalysis.AllFieldNamesOn(PivotArea.DataArea);
-
-         var pkAnalyses = _pkAnalysesTask.AggregatePKAnalysis(simulation, pkParameters, populationAnalysis.SelectedStatistics, captionPrefix[0]);
-         _allPKAnalysesIndividualPKValues.AddRange(pkAnalyses);
+         pkParametersCache.KeyValues.Each(pkParameters =>
+         {
+            var pkAnalyses = _pkAnalysesTask.AggregatePKAnalysis(simulation, pkParameters.Value, populationAnalysis.SelectedStatistics, pkParameters.Key);
+            _allPKAnalysesIndividualPKValues.AddRange(pkAnalyses);
+         });
 
          updateView();
       }
 
-
-      private IReadOnlyList<QuantityPKParameter> extractPKParameters(PopulationStatisticalAnalysis populationAnalysis, PopulationSimulation populationSimulation)
+      private Cache<string, IReadOnlyList<QuantityPKParameter>> extractPKParameters(PopulationStatisticalAnalysis populationAnalysis, PopulationSimulation populationSimulation)
       {
-         var fields = populationAnalysis.AllFields.OfType<PopulationAnalysisOutputField>().Select(x => x.QuantityPath);
-         return fields.SelectMany(x => populationSimulation.PKAnalyses.AllPKParametersFor(x)).ToList();
+         var cache = new Cache<string, IReadOnlyList<QuantityPKParameter>>();
+         var fields = populationAnalysis.AllFields.OfType<PopulationAnalysisOutputField>().ToList();
+
+         fields.Each(field => { cache.Add(field.Name, populationSimulation.PKAnalyses.AllPKParametersFor(field.QuantityPath).ToList()); });
```

Comment:
Is name path? What is the name exactly? Is there a property path ? If this is the same, use it instead because name looks like it could be duplicated

Reply:
You do not need the { } after =>

Also why the ToList?is this required

Reply:
Or maybe is name unique. I can't remember

Reply:
I don't see it being obvious. Maybe make the Cache with the PopulationAnalysisOutputField instead and then use the field.Name to get the caption

Reply:
The object itself should be unique.

Reply:
is the name unique? Probably